servicetalk-serializer-utils/src/main/java/io/servicetalk/serializer/utils/FixedLengthStreamingSerializer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.serializer.utils;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.serializer.api.SerializationException;
+import io.servicetalk.serializer.api.SerializerDeserializer;
+import io.servicetalk.serializer.api.StreamingSerializerDeserializer;
+
+import java.util.function.BiFunction;
+import java.util.function.ToIntFunction;
+import javax.annotation.Nullable;
+
+import static java.lang.Integer.BYTES;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+
+/**
+ * A {@link StreamingSerializerDeserializer} that uses a {@link SerializerDeserializer} and frames each object by
+ * preceding it with the length in bytes. The length component is fixed and always consumes 4 bytes.
+ * @param <T> The type of object to serialize.
+ */
+public final class FixedLengthStreamingSerializer<T> implements StreamingSerializerDeserializer<T> {
```

Comment:
Did you consider the option to support header/footer for the streaming approach, that would help the reading side verify that the input is indeed fixed-length and validate checksum at the end. 
Optional of course, it can still be added later and maybe even with an additional operator.